### PR TITLE
[Backport 5.4] Merge pull request #6991 from blueyed/fix-lf-cmdline-args-upstream

### PR DESCRIPTION
**Pull request by nicoddemus**
Backport of #6991

*But* I'm not so sure we should backport this, because it also contains an improvement (albeit small).

### CODE REVIEW

File: changelog/6991.improvement.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Collected files are displayed after any reports from hooks, e.g. the status from ``--lf``.
```
Comment (bluetech) [inline changelog/6991.improvement.rst:1]: Maybe we should drop this change from the backport? (the changes in `src/_pytest/terminal.py`)

File: src/_pytest/cacheprovider.py [hunk 2]
```diff
@@ -183,27 +183,35 @@ def pytest_make_collect_report(self, collector) -> Generator:
...
+                        LFPluginCollSkipfiles(self.lfplugin), "lfplugin-collskip"
+                    )
+                    self._collected_at_least_one_failure = True
+
+                session = collector.session
+                result[:] = [
+                    x
+                    for x in result
+                    if x.nodeid in lastfailed
+                    # Include any passed arguments (not trivial to filter).
```
Comment (bluetech) [inline src/_pytest/cacheprovider.py:209]: This one makes sense.

File: src/_pytest/cacheprovider.py [hunk 3]
```diff
@@ -183,27 +183,35 @@ def pytest_make_collect_report(self, collector) -> Generator:
...
+                    or session.isinitpath(x.fspath)
+                    # Keep all sub-collectors.
```
Comment (bluetech) [inline src/_pytest/cacheprovider.py:211]: But I don't really understand this one, would you mind explaining what it does?

File: src/_pytest/cacheprovider.py [hunk 4]
```diff
@@ -269,7 +277,12 @@ def pytest_collectreport(self, report):
         else:
             self.lastfailed[report.nodeid] = True
 
-    def pytest_collection_modifyitems(self, session, config, items):
+    @pytest.hookimpl(hookwrapper=True, tryfirst=True)
+    def pytest_collection_modifyitems(
+        self, config: Config, items: List[nodes.Item]
+    ) -> Generator[None, None, None]:
+        yield
```
Comment (bluetech) [inline src/_pytest/cacheprovider.py:284]: Why is this change (and in `pytest_collection_modifyitems`) needed?

Comment (RonnyPfannschmidt) [reply src/_pytest/cacheprovider.py:284]: i believe this is to ensure the actual code runs as late as possible

in this case this means, its running after most normal hook-wrappers finish (only the last bit of a earlier hook-wrapper could interact with it after that

**Comment by nicoddemus** [comment]
Closing for now, not sure we will see a 5.4.X before 6.0 anyway.

**Pull request closed without merging**